docs/developers/wallet-integration.md

```diff
@@ -0,0 +1,125 @@
+<!--
+order: 3
+-->
+
+# Wallet Integration
+
+::tip
+**Note**: want to learn more about wallet integration beyond what's covered here? Check out both the [MetaMask Wallet documentation](https://docs.metamask.io/guide/) and [Keplr Wallet documentation](https://docs.keplr.app/).
+:::
+
+Learn how to properly integrate [Metamask](https://metamask.io/) or [Keplr](https://www.keplr.app/) with a dApp on Evmos. {synopsis}
+
+## Implementation Checklist
+
+The integration implementation checklist for dApp developers consists of three categories:
+
+- implementation of [frontend features](#frontend)
+- implementation of [transactions and wallet interactions](#transactions)
+- implementation of [client-side provider](#connections)
+
+### Frontend
+
+Make sure to create a wallet-connection button for Metamask and/or Keplr on the frontend of the application. For instance, consider the "Connect to a wallet" button on the interface of [Diffusion Finance](https://app.diffusion.fi/) or the analagous button on the interface of [EvmoSwap](https://app.evmoswap.org/).
+
+### Transactions
+
+Developers enabling transactions on their dApp have to [determine wallet type](#determining-wallet-type) of the user, [create the transaction](#create-the-transaction), [request signatures](#sign-and-broadcast-the-transaction) from the corresponding wallet, and finally [broadcast the transaction](#sign-and-broadcast-the-transaction) to the network.
+
+#### Determining Wallet Type
+
+Developers should determine whether users are using Keplr or MetaMask. Whether MetaMask or Keplr is installed on the user device can be determined by checking the corresponding `window.ethereum` or `window.keplr` value.
+
+- For MetaMask: `await window.ethereum.enable(chainId);`
+- For Keplr: `await window.keplr.enable(chainId);`
+
+If either `window.ethereum` or `window.keplr` returns `undefined` after `document.load`, then MetaMask (or, correspondingly, Keplr) is not installed. There are several ways to wait for the load event to check the status: for instance, developers can register functions to `window.onload`, or they can track the document's ready state through the document event listener.
+
+After the user's wallet type has been determined, developers can proceed with creating, signing, and sending transactions.
+
+#### Create the Transaction
+
+Developers can create `MsgSend` transactions using the [evmosjs](https://github.com/evmos/evmosjs) library.
```

Comment:
nit, we have an evmosjs page in the docs but it's not currently linked. We should finish and add it to the docs